packages/legend-graph/src/graphManager/action/query/QueryUtils.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { QueryTaggedValue } from './Query.js';
+
+// NOTE: these are non-compilable profile and tag that we come up with for query
+// so that it records the class it is created from
+export const QUERY_PROFILE_PATH = 'meta::pure::profiles::query';
+export const QUERY_PROFILE_TAG_CLASS = 'class';
+
+export const createQueryClassTaggedValue = (
```

Comment:
I think this is specific to query created from data-space, can we move this to where we handle the same synthetic tagged values logic for data-space query?